doi.jq

```diff
@@ -0,0 +1,186 @@
+		"php-zendserver",
+		"phpmyadmin",
+		"postfixadmin",
+		"r-base",
+		"rocket.chat",
+		"rockylinux",
+		"sl",
+		"spiped",
+		"teamspeak",
+		"traefik",
+		"ubuntu", # direct children: buildpack-deps couchbase eclipse-temurin elasticsearch gazebo gradle ibmjava ibm-semeru-runtimes kibana kong logstash mariadb mongo neurodebian odoo php-zendserver rabbitmq ros sapmachine silverpeas swift
+		"varnish",
+		"vault",
+		"yourls",
+		"znc",
+		"adminer",
+		"aerospike",
+		"amazoncorretto", # direct children: jetty maven tomcat
+		"backdrop",
+		"buildpack-deps", # direct children: erlang gcc golang haskell haxe influxdb kapacitor node openjdk perl pypy python rakudo-star ruby rust telegraf
+		"composer", # direct children: drupal
+		"couchbase",
+		"dart",
+		"eclipse-temurin", # direct children: cassandra clojure flink gradle groovy jetty jruby lightstreamer liquibase maven neo4j orientdb solr sonarqube spark storm tomcat tomee unit zookeeper
+		"erlang", # direct children: elixir
+		"friendica",
+		"gazebo",
+		"groovy",
+		"haskell",
+		"haxe",
+		"ibm-semeru-runtimes", # direct children: maven open-liberty tomee websphere-liberty
+		"ibmjava", # direct children: maven websphere-liberty
+		"jetty", # direct children: geonetwork
+		"joomla",
+		"kong",
+		"lightstreamer",
+		"liquibase",
+		"matomo",
+		"mediawiki",
+		"monica",
+		"neurodebian",
+		"nextcloud",
+		"node", # direct children: express-gateway ghost mongo-express unit
+		"odoo",
+		"open-liberty",
+		"orientdb",
+		"python", # direct children: hylang plone satosa unit
+		"rakudo-star",
+		"ros",
+		"sapmachine", # direct children: maven
+		"satosa",
+		"silverpeas",
+		"spark",
+		"storm",
+		"swift",
+		"tomcat", # direct children: convertigo geonetwork xwiki
+		"tomee",
+		"websphere-liberty",
+		"clojure",
+		"convertigo",
+		"geonetwork",
+		"maven",
+		"plone",
+
+		empty
+	]
+;
+
+# a helper for "build_should_sbom"
+def _sbom_subset:
+	[
+		# only repositories we have explicitly verified
+		"bash",
+		"buildpack-deps",
+		"busybox",
+		"caddy",
+		"cassandra",
+		"chronograf",
+		"couchdb",
+		"drupal",
+		"eclipse-temurin",
+		"elasticsearch",
+		"flink",
+		"fluentd",
+		"gcc",
+		"ghost",
+		"gradle",
+		"haproxy",
+		"httpd",
+		"hylang",
+		"influxdb",
+		"jruby",
+		"julia",
+		"kapacitor",
+		"kibana",
+		"logstash",
+		"mariadb",
+		"memcached",
+		"mongo",
+		"mongo-express",
+		"mysql",
+		"neo4j",
+		"nginx",
+		"openjdk",
+		"perl",
+		"php",
+		"postgres",
+		"pypy",
+		"python",
+		"rabbitmq",
+		"redis",
+		"registry",
+		"rethinkdb",
+		"ruby",
+		"rust",
+		"solr",
+		"sonarqube",
+		"telegraf",
+		"tomcat",
+		"wordpress",
+		"xwiki",
+		"zookeeper",
+		empty
+	]
+;
+
+# input: "build" object (with "buildId" top level key)
+# output: boolean
+def build_should_sbom:
+	.source.arches[.build.arch].tags
+	| map(split(":")[0])
+	| unique
```

Comment:
Will this ever produce more than one value? In other words, how could a single DOI repository yield more than one repository portion of the tag?

Reply:
In theory, we could have two separate builds which reference the exact same bytes of build context with the exact same parent chain, in which case they would be joined together under a single `sourceId` (and even share `buildId`s in that case).  In practice, you're absolutely right that this will pretty much always end with an array of length one.